### PR TITLE
add Optional to getSettings and redirect from UpdateSettings to updat…

### PR DESCRIPTION
**Pull request by AdrianaSche**
…ed profile

### CODE REVIEW

File: backend/src/main/java/de/adrianaschepers/shishabuddies/model/SetupEntity.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
 import javax.persistence.Table;
 
 //was rauchst du gerade?
-//one-to-many:
+
 //ein User kann meherer setups haben
```
Comment (jamarob) [inline backend/src/main/java/de/adrianaschepers/shishabuddies/model/SetupEntity.java:11]: Please delete the comments. Better track your work in the issue itself

File: backend/src/main/java/de/adrianaschepers/shishabuddies/service/UserService.java [hunk 2]
```diff
@@ -55,7 +55,11 @@ public UserEntity createUser(UserEntity userEntity) {
     public SettingsEntity getUserSettings(UserEntity authUser) {
         Optional<UserEntity> userOpt = userRepository.findByUserName(authUser.getUserName());
         if(userOpt.isPresent()) {
-            return userOpt.get().getSettings();
+            Optional<SettingsEntity> settingsOfAuth = Optional.ofNullable(userOpt.get().getSettings());
+            if(settingsOfAuth.isPresent()){
+               return settingsOfAuth.get();
+            }
+            //return userOpt.get().getSettings();
```
Comment (jamarob) [inline backend/src/main/java/de/adrianaschepers/shishabuddies/service/UserService.java:62]:
```suggestion

```

File: backend/src/main/java/de/adrianaschepers/shishabuddies/service/UserService.java [hunk 3]
```diff
@@ -55,7 +55,11 @@ public UserEntity createUser(UserEntity userEntity) {
     public SettingsEntity getUserSettings(UserEntity authUser) {
         Optional<UserEntity> userOpt = userRepository.findByUserName(authUser.getUserName());
         if(userOpt.isPresent()) {
-            return userOpt.get().getSettings();
+            Optional<SettingsEntity> settingsOfAuth = Optional.ofNullable(userOpt.get().getSettings());
+            if(settingsOfAuth.isPresent()){
+               return settingsOfAuth.get();
+            }
```
Comment (jamarob) [inline backend/src/main/java/de/adrianaschepers/shishabuddies/service/UserService.java:61]: Why do you construct the second optional?

```suggestion
            SettinfsEntity entity = userOpt.get().getSettings();
            if(entity != null){
              return entitis;
            }
```

Imho, You gain very little by constructing the optional to use it directly after.

File: frontend/src/pages/UpdateSettings.js [hunk 4]
```diff
@@ -6,6 +6,7 @@ import { useAuth } from '../auth/AuthProvider'
 import Button from '../component/Button'
 import Page from '../component/Page'
 import Main from '../component/Main'
+import { Redirect } from 'react-router-dom'
 
 const userSettings = {
```
Comment (jamarob) [inline frontend/src/pages/UpdateSettings.js:11]: Maybe `initialSettings` as a name?

File: frontend/src/pages/UpdateSettings.js [hunk 5]
```diff
@@ -33,13 +35,17 @@ export default function UpdateSettings() {
   const handleSubmit = event => {
     event.preventDefault()
     updateSettings(newSettings, token).catch(error => console.error(error))
+    setUpdate(true)
```
Comment (jamarob) [inline frontend/src/pages/UpdateSettings.js:38]: Danger! only call `setUpdate` when the request was successful - put it inside a then call.
Maybe rename it from ùpdate` to `redirectToHome` or something?

**Review by jamarob — COMMENTED** [review]
gj 👍 left some comments

**Comment by AdrianaSche** [comment]
> gj 👍 left some comments

thx :-)

**Pull request merged**